pkg/webui/components/button/button.styl

```diff
@@ -13,69 +13,67 @@
 // limitations under the License.
 
 .button
+  // Without secondary or primary style,
+  // the button should not apply any styling.
+  // This is to cater for special situations
+  // where the button is used as a container.
   reset-button()
-  position: relative
-  display: inline-flex
-  transition: 0.2s all ease-in-out
-  border-radius: $br.l
-  outline: 0
-  cursor: pointer
-  justify-content: center
-  align-items: center
-  gap: $cs.xxs
-  height: 2.5rem
-  text-decoration: none
-  padding: 0 $cs.m
-  white-space: nowrap
+
+  &.primary, &.secondary, &.naked
+    position: relative
+    display: inline-flex
+    transition: 80ms all ease-in-out
+    border-radius: $br.m
+    outline: 0
+    cursor: pointer
+    justify-content: center
+    align-items: center
+    gap: $cs.xxs
+    height: 2.5rem
+    text-decoration: none
+    padding: 0 $cs.m
+    white-space: nowrap
+    box-sizing: border-box
 
   .arrow-icon
     transform: rotate(0deg)
-    transition: transform .2s
+    transition: transform .2s ease-in-out
+    margin-right: - $cs.xs
 
   .arrow-icon-expanded
     transform: rotate(180deg)
-    transition: transform .2s
+    transition: transform .2s ease-in-out
+
+  .icon
+    margin-left: - $cs.xxs
+
+    &:only-child
+      margin-right: - $cs.xxs
+
+  &.only-icon
+    gap: 0
 
   &.primary
-    transition: 0.2s all ease-in-out
     color: white
     background-color: $c-active-blue
 
-    &.disabled
+    &:disabled
       background-color: $c-divider-dark
 
-    &:not(.disabled)
+    &:not(:disabled)
       &:hover
-        transition: 0.2s all ease-in-out
```

Comment:
This transition prop was set at least a dozen times in various levels of the button. I don't get why, since it was already set on the base class. Let's avoid these kind of spray-can solutions.